include/onnxruntime/core/framework/float16.h

```diff
@@ -41,11 +41,10 @@ inline bool operator<(const MLFloat16& left, const MLFloat16& right) { return le
 // BFloat16
 struct BFloat16 {
   uint16_t val{0};
-#if defined(USE_ROCM)
-  ORT_HOST_DEVICE BFloat16() = default;
-#else
-  BFloat16() = default;
+#if defined(__clang__) && defined(__HIP__)
```

Comment:
Is it not sufficient to just check for `__HIP__`?

Reply:
Looks like you're right.  `__HIP__` should be sufficient, based on https://github.com/ROCm-Developer-Tools/HIP/blob/develop/docs/markdown/hip_porting_guide.md#compiler-defines-summary.